python/pyiceberg/catalog/rest.py

```diff
@@ -0,0 +1,373 @@
+        return ConfigResponse(**response.json())
+
+    def _split_namespace_and_table(self, identifier: Union[str, Identifier]) -> Properties:
+        identifier = self.identifier_to_tuple(identifier)
+        return {"namespace": NAMESPACE_SEPARATOR.join(identifier[:-1]), "table": identifier[-1]}
+
+    def _split_namespace_and_name(self, identifier: Union[str, Identifier]) -> Dict[str, Union[Tuple[str, ...], str]]:
+        identifier = self.identifier_to_tuple(identifier)
+        return {"namespace": identifier[:-1], "name": identifier[-1]}
+
+    def _handle_non_200_response(self, exc: HTTPError, error_handler: Dict[int, Type[Exception]]):
+        try:
+            response = ErrorResponse(**exc.response.json())
+        except Exception:  # pylint: disable=W0012,W0703
+            # In the case we don't have a proper response
+            response = ErrorResponse(
+                error=ErrorResponseMessage(
+                    message=f"Could not decode json payload: {exc.response.text}",
+                    type="RESTError",
+                    code=exc.response.status_code,
+                )
+            )
+
+        code = exc.response.status_code
+        if code in error_handler:
+            raise error_handler[code](response) from exc
+        elif code == 400:
+            raise BadRequestError(response.error.message) from exc
+        elif code == 401:
+            raise UnauthorizedError(response.error.message) from exc
+        elif code == 503:
+            raise ServiceUnavailableError(response.error.message) from exc
+        elif 500 <= code < 600:
+            raise ServerError(response.error.message) from exc
+        else:
+            raise RESTError(response.error.message) from exc
+
+    def create_table(
+        self,
+        identifier: Union[str, Identifier],
+        schema: Schema,
+        location: Optional[str] = None,
+        partition_spec: PartitionSpec = UNPARTITIONED_PARTITION_SPEC,
+        sort_order: SortOrder = UNSORTED_SORT_ORDER,
+        properties: Optional[Properties] = None,
+    ) -> RestTable:
+        namespace_and_table = self._split_namespace_and_table(identifier)
+        properties = properties or {}
+        request = CreateTableRequest(
+            name=namespace_and_table["table"],
+            location=location,
+            table_schema=schema,
+            partition_spec=partition_spec,
+            write_order=sort_order,
+            properties=properties,
+        )
+        serialized_json = request.json()
+        response = requests.post(
+            self.url(Endpoints.create_table, namespace=namespace_and_table["namespace"]),
+            data=serialized_json,
+            headers=self.headers,
+        )
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {409: TableAlreadyExistsError})
+
+        return RestTable(identifier=identifier, **response.json())
```

Comment:
`identifier` should include the catalog name. Otherwise you can't tell where the table was loaded from if there are multiple catalogs.

Reply:
Ah, wasn't aware of that. Added those to create, load and list tables

Reply:
This feels slightly weird to me. Thinking of the CLI, should we also return a name when printing them to stdout?

Reply:
Actually, it is just the table's identifier that includes catalog name. When accepting or producing an identifier, the catalog's name will not be present. If you look up `db` in some catalog instance and see `t1` and `t2`, you don't want to strip the catalog name off of the returned result. So `[db, t1]` and `[db, t2]` are correct API results.

But when we create a table, we pass in a string version of the identifier (at least, in Java) so that the table can be identified by name in logs or other situations. It is reasonable to track that as an identifier here, but in that context I think we want to preserve information about the catalog that loaded the table. That's why I think we should add it here, but not elsewhere.

Reply:
I can see that, but the catalog name is arbitrary and just passed in through the constructor of the catalog. I agree that we want to return the fully qualified name, including the database/schema.